src/app/services/alert.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import {Observable, Subject} from 'rxjs';
+import {NavigationStart, Router} from '@angular/router';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AlertService {
+  private subject = new Subject<any>();
+  private keepAfterNavigationChange = false;
+
+  constructor(private router: Router) {
+    // clear alert message on route change
+    router.events.subscribe(event => {
+      if (event instanceof NavigationStart) {
+        if (this.keepAfterNavigationChange) {
+          // only keep for a single location change
+          this.keepAfterNavigationChange = false;
+        } else {
+          // clear alert
+          this.subject.next();
+        }
+      }
+    });
+  }
+  success(message: string, keepAfterNavigationChange = false) {
```

Comment:
method naming should tell about some action or etc.. Also I don't know is it a good idea for the second parameter. Why don't you just set `this.keepAfterNavigationChange = false` inside method (27 line)?

Reply:
theyre left so that the variable is false by default, but if we want the alert to stay up because for example we are redirecting the user to another page, we call this function with variable true so that it stays visible

Reply:
ok, cool